libraries/ms-common/inc/adc.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+// Analog to Digital Converter HAL Inteface
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "status.h"
+
+typedef enum {
+  ADC_MODE_SINGLE = 0,
+  ADC_MODE_CONTINUOUS,
+} ADCMode;
+
+typedef enum {
+  ADC_CHANNEL_0 = 0,
+  ADC_CHANNEL_1,
+  ADC_CHANNEL_2,
+  ADC_CHANNEL_3,
+  ADC_CHANNEL_4,
+  ADC_CHANNEL_5,
+  ADC_CHANNEL_6,
+  ADC_CHANNEL_7,
+  ADC_CHANNEL_8,
+  ADC_CHANNEL_9,
+  ADC_CHANNEL_10,
+  ADC_CHANNEL_11,
+  ADC_CHANNEL_12,
+  ADC_CHANNEL_13,
+  ADC_CHANNEL_14,
+  ADC_CHANNEL_15,
+  ADC_CHANNEL_TEMP,
+  ADC_CHANNEL_REF,
+  ADC_CHANNEL_BAT,
+  NUM_ADC_CHANNEL
+} ADCChannel;
+
+typedef void (*ADCCallback)(ADCChannel adc_channel, uint16_t reading, void *context);
```

Comment:
I think it'd be a good idea to make it obvious whether this is returning the raw value or voltage.